sugondat/chain/node/src/proposer.rs

```diff
@@ -89,8 +90,58 @@ impl<P: ProposerInterface<Block> + Send> ProposerInterface<Block> for BlockLimit
             // testing for detection of healthiness.
             parent_header.number == 0
         };
+        let exceeded_max_skipped_blocks = {
+            let maybe_last_relay_block_number: Option<u32> =
+                inherent_digests.logs.iter().find_map(|log| {
+                    sp_tracing::info!(?log, "SKIP");
+                    match log {
+                        DigestItem::Other(ref bytes) if bytes.starts_with(b"srbn") => {
+                            sp_tracing::info!(?bytes, "SKIP");
+                            Decode::decode(&mut &bytes[4..]).ok()
+                        }
+                        DigestItem::Other(ref bytes) if bytes.starts_with(b"snmt") => {
+                            sp_tracing::info!("SKIP nmt root");
+                            None
+                        }
+                        _ => None,
+                    }
+                });
+            let relay_block_number = paras_inherent_data.validation_data.relay_parent_number;
 
-        if has_downward_message || has_go_ahead || has_transactions || first_block {
+            sp_tracing::info!(?maybe_last_relay_block_number, "SKIP");
+            sp_tracing::info!(?relay_block_number, "SKIP");
```

Comment:
`sp_tracing` should be used only in the runtime. I believe that regular tracing is preferable on the node-side.

Reply:
I have switched to `log` in the node and `sp_tracing` in the runtime, but logging from the runtime continues to not work